docs/proposals/deletions_object_storage.md

```diff
@@ -0,0 +1,100 @@
+---
+title: Delete series for object storage
+type: proposal
+menu: proposals
+status: approved
+owner: bwplotka, Harshitha1234, metalmatze
+---
+
+### Ticket: https://github.com/thanos-io/thanos/issues/1598
+## Summary
+
+This design document proposes deletion of series for object storage in Thanos. This feature mainly causes changes in the Compactor and Store components and the object storage itself where the changes are expected to be reflected.
+
+## Motivation
+
+The main motivation for considering deletions in the object storage are the following use cases
+
+*   **Accidental insertion of confidential data:** This is a scenario where a user accidentally inserts confidential data and wants to delete it immediately. In this case, the user expects their request to accelerate an immediate deletion of the series pertaining to the blocks concerned with that specific data to be deleted.
+*   **GDPR:** Masking data and eventual deletion is expected.
+*   **Deletions to sustain user requirements:** Let’s assume the user has some data which leads to some unexpected results or causes performance degradation (due to high cardinality) and the user wants to restore the previous data set-up for obtaining the desired results. In this scenario, the user would want to send a request to mask the data for the time being as there isn’t a high priority requirement to delete the data but eventually during the compaction the user can expect the data to be deleted by the compactor not leading to any major performance issues.
+*   **Helps achieving series based retention (e.g rule aggregation) [#903](https://github.com/thanos-io/thanos/issues/903).**
+
+## Goals
+
+*   Unblock users and allow series deletion in the object storage using tombstones.
+*   Deal with compaction and downsampling of blocks with tombstones.
+*   Performing deletions at admin level.
+
+## Proposed Approach
+
+*   We propose to implement deletions via the tombstones approach using a CLI tool.
+*   A tombstone is proposed to be a **Custom format global - single file per request**.
```

Comment:
I haven't found the exact format of this file documented. Anyway, I would add the timestamp of the deletion request inside as well. Would allow to build hard deletion grace periods, without having to read it from the object storage (object creation).

Reply:
still missing @Harshitha1234